bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,114 @@
+from geopy.geocoders import Nominatim
+from timezonefinder import TimezoneFinder
+from pytz import country_timezones, timezone
+from datetime import datetime
+from discord.ext.commands import Cog, command
+from requests import get
+from discord import Embed
+from string import capwords
+
+
+class WeatherCog(Cog, name="Weather", description="get current weather information from a city"):
+    def __init__(self, bot):
+        self.bot = bot
+    
+    @command(name='weather', help='Show weather information in your city', usage="{city}")
+    async def weather(self, ctx, *city_input):
+        city = capwords(" ".join(city_input))
+        
+        # get current date and time from the city
+        timezone_city = getTimezone(city)
+
+        api_key = "441df3a5cadc2498e093c0367cae6817"
+        # complete_url to retreive weather info
+        complete_url = f"http://api.openweathermap.org/data/2.5/weather?appid={api_key}&q={city}"
```

Comment:
### Separating the part that fetches the weather

This could also be separated the command. You could have an async function called `get_weather` in which you would pass the formatted city name as the parameter and that would return the json if it  worked or None if it didn't or something like that. 

There's multiple reason why it would be particularity interesting to do that. By separating this, we can first reuse it elsewhere in the class super easily plus if we bring changes, everyone will have those same changes. We also  abstract the command from the data source which means if we decide to make the data come from a database instead or a new API, all we have to do is change this function and the command will stay the same or almost. We could go a step further to really make this more usable but it would be overkill.

### Static attribute for the base url

Also, as I said in another comment, I would pull out the base url (`http://api.openweathermap.org/data/2.5/`) from the method and create a constant called `OPENWEATHER_BASE_URL` or `BASE_URL`. 

Imagine you added a command for getting the forcast, solar radiation, global weather alert, road risk, etc. We don't want to repeat the base url because if the base url changes, for example we want to pass to version 2.6 of the API, we'll need to go through each commands or methods using this url and change it. 

### Why am I so annoying?

Because it would be cleaner and more reusable for futur features and thinking that way is a good habit to have.